node/node.go

```diff
@@ -1442,7 +1388,7 @@ func (n *Node) getAccountRootHashAndVal(address []byte, accBytes []byte, key []b
 }
 
 func (n *Node) getProof(rootHash []byte, key []byte) (*common.GetProofResponse, error) {
-	tr, err := n.stateComponents.AccountsAdapter().GetTrie(rootHash)
+	tr, err := n.stateComponents.AccountsAdapterAPI().GetTrie(rootHash)
```

Comment:
should call the recreate trie method

Maybe add a TODO to not leave this mess and implement that accounts DB wrapper we've talked about